guardian/mobile-save-for-later
# Remove superfluous resources - these have been replaced by GuCDK versions

## What does this change?

We are migrating the `mobile-save-for-later` service from CloudFormation to `@guardian/cdk`. The migration will involve 5 PRs:

1. Add `@guardian/cdk` to repository; update CI & deployment wiring accordingly (https://github.com/guardian/mobile-save-for-later/pull/64)
2. Provision a new version of the infrastructure using `@guardian/cdk` (https://github.com/guardian/mobile-save-for-later/pull/65)
3. Move DNS configuration from CloudFormation into `@guardian/cdk` (but continue routing requests to old infrastructure) (https://github.com/guardian/mobile-save-for-later/pull/66)
4. Update DNS configuration (start routing requests to new infrastructure) (https://github.com/guardian/mobile-save-for-later/pull/67)
5. **Remove old infrastructure [this PR]**

Note that after merging this PR we will still have [a handful of resources in CFN](https://github.com/guardian/mobile-save-for-later/blob/489fe9ebaf6a090ef64c4e5c25d003713dee4736/mobile-save-for-later/conf/cfn.yaml) (and therefore we will still need to [import the CFN file](https://github.com/guardian/mobile-save-for-later/blob/489fe9ebaf6a090ef64c4e5c25d003713dee4736/cdk/lib/mobile-save-for-later.ts#L27-L34) via `cdk`):

- `AWS::CloudWatch::Alarm` x 2
- `AWS::DynamoDB::Table`

I suggest that we deal with these in a couple of future PRs (since it's not quite as straightforward as deleting things):

1. Remove the alarms once https://theguardian.atlassian.net/browse/LIVE-4204 is complete
2. Port the Dynamo table from CFN to TypeScript (this will be similar to https://github.com/guardian/presence-indicator/pull/129 and [these docs](https://github.com/guardian/cdk/blob/main/docs/stateful-resources.md) are likely to provide some helpful context)

## How to test

I've deployed this to `CODE` and confirmed that the service still works as expected (i.e. you can still save/view saved articles).

*N.B. this change must be deployed using Riff-Raff's 'dangerous mode' in order to bypass a safety mechanism designed to prevent the accidental deletion of stateful resources. For background on this, [see this email](https://groups.google.com/a/guardian.co.uk/g/engineering/c/AsLBhAIX1oA/m/0FWu8dfGBgAJ).*

## How can we measure success?

We will no longer have duplicate resources in AWS.

## Have we considered potential risks?

Yes, the main risk here is that we've accidentally deleted a resource that we still need! I think the testing performed is sufficient mitigation for this risk.

Once this PR is merged it would also be quite difficult to rollback (since we'd have to reprovision everything, as well as updating DNS). However, the [dashboard for this service](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#dashboards:name=SaveForLater) confirms that all traffic is now being handled via `cdk`-defined infrastructure and the error/latency metrics are all looking good (in fact they are looking slightly better since the migration!). Consequently, I think it's the right time to tidy this up.
